connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java

```diff
@@ -138,6 +138,11 @@ private synchronized void sendRecords(List<SourceRecord> records) {
                         @Override
                         public void onCompletion(RecordMetadata recordMetadata, Exception e) {
                             if (e != null) {
+                                // Given the default settings for zero data loss, this should basically never happen --
+                                // between "infinite" retries, indefinite blocking on full buffers, and "infinite" request
+                                // timeouts, callbacks with exceptions should never be invoked in practice. If the
+                                // user overrode these settings, the best we can do is notify them of the failure via
+                                // logging.
                                 log.error("Failed to send record: ", e);
```

Comment:
Would be very helpful for users to get some information on which record. At the very least which topic, but perhaps key / value too (at "debug" level perhaps)